### DISCUSSION TITLE
Conditional question: error when trying to get data

### DISCUSSION THREAD

**Discussion by lhunim95** [jspsych/jsPsych#2362] [Q&A] [answered]
Dear community,

I am currently trying to create a filter question from a survey with multiple questions. If this particular question is answered "No", I would like to ask another question afterwards. If the question is answered with "Yes", the timeline should simply continue. However, I have the problem that the following error message always appears: 
"cannot read properties of undefined (reading responses)". 

Filter question is Question 3 and the conditional timeline should run when the question is answered with "no" (= value 1).

```
var survey_one = {
  type: 'survey-likert',
  questions: [
    {prompt: "Question1", labels: scale_5, name: 'question1', required: true},
    {prompt: "Question2", labels: scale_3, name: 'question2', required: true},
    {prompt: "Question3", labels: scale_4, name: 'question3', required: true}
  ],
  button_label: 'Weiter'
}
timeline.push(survey_one)

var conditional_question = {
  type: 'survey-likert',
  questions: [
    {prompt: "conditional", labels: scale_6, name: 'conditional_question', required: true}
  ],
  button_label: 'Weiter'
}

var conditional = {
  timeline: [conditional_question],
  conditional_function() {
    var last_trial_data = jsPsych.data.get().filter({name: 'question3'}).values()[0].responses
    if (last_trial_data.response == 1) {
      return true;
    } else {
      return false;
    }
  }
};
timeline.push(final_questions_conditional)
```

So something seems to be wrong with accessing data, but I do not know how to solve it. 

I'd be very happy about some help. :)

Best regards, 
Larissa

**Comment by jodeleeuw** [comment, marked as answer]
Hi @lhunim95,

The line where you are filtering the data isn't quite right. Unfortunately you can't filter on the `name` property of the question, because that data is nested inside the `response` property. `filter()` also only works at the whole-trial level, so you can't filter to just one of the questions.

Try this instead:

```js
var last_trial_data = jsPsych.data.get().last(1).values()[0].response.question3;
if(last_trial_data == 1) {
 ...
```

This uses the `last()` function to get only the previous trial, and then accesses the `question3` answer in the `response` data. Note that this is `response` and not `responses`. This was changed in v6.3 for better consistency across plugins.

**Reply by lhunim95** [reply]
you made my day!!! Thank you so much :)
